csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests.Capi/C_Api_Sample.cpp

```diff
@@ -0,0 +1,149 @@
+    const OrtTensorTypeAndShapeInfo* tensor_info = OrtCastTypeInfoToTensorInfo(typeinfo);
+    ONNXTensorElementDataType type = OrtGetTensorElementType(tensor_info);
+    printf("Input %d : type=%d\n", i, type);
+
+    // print input shapes/dims
+    size_t num_dims = OrtGetNumOfDimensions(tensor_info);
+    printf("Input %d : num_dims=%zu\n", i, num_dims);
+    input_node_dims.resize(num_dims);
+    OrtGetDimensions(tensor_info, (int64_t*)input_node_dims.data(), num_dims);
+    for (int j = 0; j < num_dims; j++)
+      printf("Input %d : dim %d=%jd\n", i, j, input_node_dims[j]);
+
+    OrtReleaseTypeInfo(typeinfo);
+  }
+  OrtReleaseAllocator(allocator);
+
+  // Results should be...
+  // Number of inputs = 1
+  // Input 0 : name = data_0
+  // Input 0 : type = 1
+  // Input 0 : num_dims = 4
+  // Input 0 : dim 0 = 1
+  // Input 0 : dim 1 = 3
+  // Input 0 : dim 2 = 224
+  // Input 0 : dim 3 = 224
+
+  //*************************************************************************
+  // Similar operations to get output node information.
+  // Use OrtSessionGetOutputCount(), OrtSessionGetOutputName()
+  // OrtSessionGetOutputTypeInfo() as shown above.
+
+  //*************************************************************************
+  // Score the model using sample data, and inspect values
+
+  size_t input_tensor_size = 224 * 224 * 3;  // simplify ... using known dim values to calculate size
+                                             // use OrtGetTensorShapeElementCount() to get official size!
+
+  std::vector<float> input_tensor_values(input_tensor_size);
+  std::vector<const char*> output_node_names = {"softmaxout_1"};
+
+  // initialize input data with values in [0.0, 1.0]
+  for (unsigned int i = 0; i < input_tensor_size; i++)
+    input_tensor_values[i] = (float)i / (input_tensor_size + 1);
+
+  // create input tensor object from data values
+  OrtAllocatorInfo* allocator_info;
+  CHECK_STATUS(OrtCreateCpuAllocatorInfo(OrtArenaAllocator, OrtMemTypeDefault, &allocator_info));
+  OrtValue* input_tensor = NULL;
+  CHECK_STATUS(OrtCreateTensorWithDataAsOrtValue(allocator_info, input_tensor_values.data(), input_tensor_size * sizeof(float), input_node_dims.data(), 4, ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT, &input_tensor));
+  assert(OrtIsTensor(input_tensor));
+  OrtReleaseAllocatorInfo(allocator_info);
+
+  // score model & input tensor, get back output tensor
+  OrtValue* output_tensor = NULL;
+  CHECK_STATUS(OrtRun(session, NULL, input_node_names.data(), (const OrtValue* const*)&input_tensor, 1, output_node_names.data(), 1, &output_tensor));
+  assert(OrtIsTensor(output_tensor));
+
+  // Get pointer to output tensor float values
+  float* floatarr;
+  OrtGetTensorMutableData(output_tensor, (void**)&floatarr);
+  assert(abs(floatarr[0] - 0.000045) < 1e-6);
+
+  // score the model, and print scores for first 5 classes
+  for (int i = 0; i < 5; i++)
+    printf("Score for class [%d] =  %f\n", i, floatarr[i]);
```

Comment:
Compare the values?

Reply:
The first value is compared (in the assert statement, as a test-check). For floats, they have have to be compared with some tolerances -- since this snippet doubles as a user sample, it may add more code bloat.